src/wallet.test.ts

```diff
@@ -326,6 +331,68 @@ describe('wallet', () => {
     });
   });
 
+  describe('signTypedDataV3', () => {
+    it('should sign data and normalizes verifyingContract', async () => {
+      const { engine } = createTestSetup();
+      const getAccounts = async () => testAddresses.slice();
+      const witnessedMsgParams: TypedMessageParams[] = [];
+      const processTypedMessageV3 = async (msgParams: TypedMessageParams) => {
+        witnessedMsgParams.push(msgParams);
+        // Assume testMsgSig is the expected signature result
+        return testMsgSig;
+      };
+
+      engine.push(
+        createWalletMiddleware({ getAccounts, processTypedMessageV3 }),
+      );
+
+      const message = {
+        types: {
+          EIP712Domain: [
+            { name: 'name', type: 'string' },
+            { name: 'version', type: 'string' },
+            { name: 'chainId', type: 'uint256' },
+            { name: 'verifyingContract', type: 'address' },
+          ],
+        },
+        primaryType: 'EIP712Domain',
+        domain: {
+          verifyingContract: '996101235222674412020337938588541139382869425796',
+        },
+        message: {},
+      };
+
+      const stringifiedMessage = JSON.stringify(message);
+      const expectedStringifiedMessage = JSON.stringify({
+        ...message,
+        domain: {
+          verifyingContract: '0xae7ab96520de3a18e5e111b5eaab095312d7fe84',
+        },
+      });
+
+      const payload = {
+        method: 'eth_signTypedData_v3',
+        params: [testAddresses[0], stringifiedMessage], // Assuming testAddresses[0] is a valid address from your setup
+      };
+
+      const signTypedDataV3Response = await pify(engine.handle).call(
+        engine,
+        payload,
+      );
+      const signTypedDataV3Result = signTypedDataV3Response.result;
+
+      expect(signTypedDataV3Result).toBeDefined();
+      expect(signTypedDataV3Result).toStrictEqual(testMsgSig);
+      expect(witnessedMsgParams).toHaveLength(1);
+      expect(witnessedMsgParams[0]).toMatchObject({
+        from: testAddresses[0],
+        data: expectedStringifiedMessage,
+        version: 'V3',
+        signatureMethod: 'eth_signTypedData_v3',
+      });
+    });
+  });
```

Comment:
[suggestion] could include tests for:
- octal addresses
- inputs that are not parsable

Reply:
As mentioned above (https://github.com/MetaMask/eth-json-rpc-middleware/pull/309#discussion_r1651252283), these test cases are already covered by the normalizer unit tests. I added this single eth_signTypedData_v3 test because it was missing from the wallet tests. Is there any advantage to adding essentially the same tests here as well?